### PR TITLE
Implement getUser Resolver and Unit Tests

### PR DESCRIPTION
**Pull request by Tungaamaa**
****Description****
This PR adds the implementation and unit tests for the getUser GraphQL resolver. The resolver retrieves user details based on the provided id, ensuring proper error handling for invalid or missing users.

**Changes Made**
Implemented getUser resolver in resolvers/queries.ts
Added validation to check if the provided id is a valid MongoDB ObjectId
Ensured proper error handling for invalid IDs, non-existent users, and database failures
Created get-user.spec.ts to test various scenarios, including:
Valid user retrieval
Handling of invalid user IDs
Case where the user is not found
Database failure handling
**How to Test**
Run unit tests with jest to verify functionality:
yarn nx test get-user.spec.ts
Ensure all tests pass and cover edge cases.
**Additional Notes**
Mocked User.findById to simulate database responses.
Used catchError for consistent error handling.

### CODE REVIEW

**Review by erik-tsogt — DISMISSED** [review]
LGTM

File: apps/L1US/chatbot/backend/src/schemas/user.schema.ts [hunk 1]
```diff
@@ -37,4 +37,8 @@ export const UserTypeDefs = gql`
     registerUser(input: RegisterUserInput!): RegisterUserResponse!
     loginUser(input: LoginInput!): LoginUserResponse!
   }
+
+  type Query {
+    getUser(id: ID!): User!
```
Comment (DuurenbayarUlz) [inline apps/L1US/chatbot/backend/src/schemas/user.schema.ts:42]: userId: ID!

File: apps/L1US/chatbot/backend/src/resolvers/queries/get-user.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { QueryResolvers } from '../../generated';
+import { User } from '../../models';
+import { catchError } from '../../utils';
+import mongoose from 'mongoose';
+
+export const getUser: QueryResolvers['getUser'] = async (_, { id }) => {
```
Comment (DuurenbayarUlz) [inline apps/L1US/chatbot/backend/src/resolvers/queries/get-user.ts:6]: userId

**Review by Tungaamaa — COMMENTED** [review]
changed id to userID

**Review by DuurenbayarUlz — APPROVED** [review]

**Review by erik-tsogt — APPROVED** [review]
